digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressLinkBuilder.scala

```diff
@@ -110,14 +110,9 @@ object RoadAddressLinkBuilder extends AddressLinkBuilder {
     val headAddress = roadAddresses.headOption
     val geom = GeometryUtils.truncateGeometry3D(roadLink.geometry, 0.0, roadLink.length)
     val length = GeometryUtils.geometryLength(geom)
-    val sideCode = if (roadLink.trafficDirection == TrafficDirection.TowardsDigitizing) {
-      SideCode.TowardsDigitizing
-    } else if (roadLink.trafficDirection == TrafficDirection.AgainstDigitizing) {
-      SideCode.AgainstDigitizing
-    } else if (roadLink.trafficDirection == TrafficDirection.BothDirections) {
-      SideCode.BothDirections
-    } else {
-      SideCode.Unknown
+    val sideCode =headAddress match {
+      case Some(road) => road.sideCode
+      case _ => SideCode.Unknown
```

Comment:
in case there is no address(es), shouldnt we then get the sideCode from roadLinks?

Reply:
vvhroadlink doesnt have sidecode. It has trafic direction but according to JanneG it is very unreliable and it is better to have unknown than bad data